File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPHeldItemSlot.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPHeldItemSlot implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "held_item_slot";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 0;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getInventory().getHeldItemSlot();
    }

    @Override
    public void restoreProperty(Player player, Integer slot) {
        player.getInventory().setHeldItemSlot(slot);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPHeldItemSlot.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPHeldItemSlot extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "held_item_slot"

  override def getDefaultSerializedProperty: Integer = 0

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getInventory.getHeldItemSlot

  override def restoreProperty(player: Player, slot: Integer): Unit =
    player.getInventory.setHeldItemSlot(slot)